### ISSUE TITLE
Question for drawing the TSNE graph

### ISSUE THREAD

**Issue by beinathan** [aerorobotics/neural-fly#6]
Hello! Thank you for sharing your thoughts. I have a question for you, can you tell me if the data for the linear weights (a*) is obtained after many iterations of training when you plot the TSNE graph, I tried to reproduce it with the sample code you provided but I didn't get a convergence result. I would be grateful if you could reply.

**Comment by AndersonRayner** [comment]
@mtoc12

**Comment by Beethoven-Q** [comment]
I am also confused how to reproduce the t-SNE plot in the paper. That looks amazing and perfect but I failed to reproduce it with the given data and code. I was using TSNE from sklearn.manifold. The training_and_validation code works well and converge to a low loss level. Then I used the saved model at each training stage, i.e. the phi_net, to calculate the corresponding a*. Then I flattened each a* (into a 1D vector) for t-SNE plotting. But the t-SNE plot doesn't seem to cluster well. Even after I tuned the parameter for t-SNE, I still cannot get a well clustered t-SNE like in the paper. I would appreciate if anyone can explain more about how to reproduce the t-SNE plot.
Thank you!

**Comment by rdesc** [comment]
Any updates on this?
